include/xtensor/xchunked_array.hpp

```diff
@@ -0,0 +1,91 @@
+#include <vector>
+#include <array>
+#include "xarray.hpp"
+
+template <class Chunk_type>
+class xchunked_array
+{
+    public:
+
+        using const_reference = typename Chunk_type::const_reference;
+
+        template <class... Args>
+        inline const_reference operator()(Args... args) const
+        {
+            return access_impl(args...);
+        }
+
+        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunks):
+            m_shape(shape),
+            m_chunk_shape(chunks)
+        {
+            std::vector<size_t> shape_chunk;
+            size_t di = 0;
+            std::cout << "Creating chunked array with shape";
+            for (auto s: shape)
+            {
+                std::cout << " " << s;
+                size_t chunk_nb = s / chunks[di];
+                if (s % chunks[di] > 0)
+                    chunk_nb += 1;  // edge chunk
+                shape_chunk.push_back(chunk_nb);
```

Comment:
`shape_chunk` should be initialized to `shape.size()`. You can then work on `shape_chunk[i]` and avoid `push_back` (which may trigger dynamic allocation).

Reply:
:+1: